### ISSUE TITLE
Multiple select height issue

### ISSUE THREAD

**Issue by xenoborg** [select2/select2#4756]
When using a multi select, I  can't properly change the height.

.select2-container--default .select2-selection--multiple {
.select2-container .select2-selection--multiple {
min-height: ;
}
doesn't go below its default 32px; when changed to anything lower ie 26px;
changing all margin-top to 0px; and removing all padding from the stylesheet doesn't help. and I use a reset stylesheet.

Not being able to change the height I have to align the selected grey boxes and the input height to vertical align with margin-top, but this always results in extra "padding" on the bottom, adding anything bigger than margin-top: 5px; and the select box gets bigger, meaning the padding still stays.

tldr; min-height does nothing, and centering the crap inside still results in extra "padding-bottom" and the select2.css stylesheet is an unfathomable nightmare.

Would just like a 26px height multi select box, with the selected options being 22 - 24px height and vertical centered.

**Comment by korent** [comment]
If you want to disable the "extra" height (the 7px) select2 adds when using multi-select option I think this css might do the trick and it won't mess up the style of your dropdown list. Took me a while to figure it out.

.select2-container--open .select2-selection__rendered{
    border: none;
}

.select2-container .select2-search--inline{
    display: none;
}

**Comment by abartier** [comment]
The only way I found is to set something like:

**Comment by leahwchang** [comment]
@korent Hi! Your code worked for me but because of display: none, the placeholder text is not showing. I tried to add it to data-placeholder with jQuery, but initiating select2 seems to ignore prior javascript set on the select input... any ideas to show the placeholder text?

**Comment by khalyomede** [comment]
Can reproduce the issue with this [JSFiddle](https://jsfiddle.net/8x4tjbmr/). Do you know if you plan on fixing this issue?

**Comment by buttflattery** [comment]
up

**Comment by Codepwned2** [comment]
FOUND CSS SOLUTION:
The issue is with 
.select2-container .select2-search--inline .select2-search__field {
    box-sizing: border-box;
    border: none;
    font-size: 100%;
    margin-top: 5px; <!---- THAT GUY
    padding: 0;
}

TO FIX:
Simple set it to 0px. Very odd behavior. Works in Chrome, Firefox, Safari (latest versions on windows)

**Comment by mtngnn** [comment]
Codepwned2  has the best answer. It is working like a charm now.

**Comment by RoyBellingan** [comment]
I think the Codepwned2 answer does not apply any longer.

**Comment by JVwork** [comment]
So... I've come across this today and the extra 7px are (in my opinion) there to compensate for `margin-top` (`5px`) and horizontal `border`s (`1px` each) on `select2-selection__choice` list items.
My solution:

```
.select2-container .select2-selection--multiple .select2-selection__rendered {
    display: flex;
    flex-wrap: wrap;
    gap: 5px;
}

.select2-container .select2-selection--multiple .select2-selection__rendered li {
    margin: 0;
}
```

... of course adjust the classes to adapt to your specific case (like `.select2-container--default`, etc, if needed)
Hope it helps :)

**Comment by imskriaz** [comment]
I have fixed it with this 

body .select2-container .select2-selection--multiple .select2-selection__rendered {
    display: inline;
}

body .select2-container--default .select2-selection--multiple .select2-selection__rendered li {
    margin: 3px;
}

body .select2-container--default .select2-search--inline .select2-search__field {
    margin: 0;
    max-height: 1em;
    padding: calc(1em - 2px);
}

.select2-container {
    min-width: 100%;
    max-width: 100%;
    border: 1px solid #aaa;
    border-radius: 4px;
}

.select2-container--default.select2-container--focus .select2-selection--multiple, .select2-container--default .select2-selection--multiple {
    border: none;
}
